lib/pbench/server/api/resources/query_apis/month_indices.py

```diff
@@ -27,23 +27,13 @@ def __init__(self, config: PbenchServerConfig, logger: Logger):
 
     def get(self):
         """
-        Report the month suffixes existing for run data indices.
+        Report the month suffixes for run data indices.
 
-        Required headers include
+        NOTE: No authorization or input parameters are required for this API.
```

Comment:
This feels odd.  It seems like an information leak.

For instance, it's conceivable that someone could reach unwarranted conclusions about Red Hat business by observing when we produce performance results and when we don't.

I don't know whether we should actually worry about this, since the resolution of the results is so low (a single benchmark result would be enough to cause its production month to be returned in the list) and, once we support a significant number of clients, there will never be a month when there are _no_ results, at which point there probably won't be anything to be gained from the information returned by this request (although, at that point, perhaps this request will no longer even be necessary...).

Reply:
If we're going to support the normal starting dashboard page without login, this needs to work because it's how the dashboard sets up the date picker. The alternative is to allow no access at all to a user who's not logged in. (Something I'd personally prefer... but it would take a lot more work on the dashboard side to _only_ support our currently hypothetical share URIs for sessions that aren't logged in.)

Reply:
> If we're going to support the normal starting dashboard page without login, this needs to work

Fair enough.

> The alternative is to allow no access at all to a user who's not logged in. (Something I'd personally prefer... [...])

I'm not opposed to that, but I understand that "we" think that it would be a Good Thing™ not to have that hurdle for viewers.

> it would take a lot more work on the dashboard side to only support our currently hypothetical share URIs for sessions that aren't logged in.

I'm _hoping_ that the hypothetical share URIs are basically orthogonal to the other issue.  That is, I expect that the share URIs are basically a content manifest; whether the requestor can access that content is a separate question.  (And (I think) that question gets answered on the server side, not the client side -- the client just makes the request(s) and displays what the server returns, which might be a combination of content and access errors -- and this is true regardless of whether the user is logged in or not...but, maybe I'm missing something.)

Reply:
> I expect that the share URIs are basically a content manifest; whether the requestor can access that content is a separate question. (And (I think) that question gets answered on the server side, not the client side -- the client just makes the request(s) and displays what the server returns, which might be a combination of content and access errors -- and this is true regardless of whether the user is logged in or not...but, maybe I'm missing something.)

That's how I see it, too. We'll have to see how to get there from where we are.